examples/conditionals.yaml

```diff
@@ -1,35 +1,41 @@
 # Conditionals provide a way to affect the control flow of a
 # workflow at runtime, depending on parameters. In this example
-# the 'print-hello' template may or may not be executed depending
-# on the input parameter, 'should-print'. When submitted with:
+# the 'printHello' template may or may not be executed depending
+# on the input parameter, 'shouldPrint'. When submitted with:
 #   argo submit examples/conditionals.yaml
-# the step will be skipped since 'should-print' will evaluate false.
+# the step will be skipped since 'shouldPrint' will evaluate false.
 # When submitted with:
-#   argo submit examples/conditionals.yaml -p should-print=true
-# the step will be executed since 'should-print' will evaluate true.
+#   argo submit examples/conditionals.yaml -p shouldPrint=true
+# the step will be executed since 'shouldPrint' will evaluate true.
 apiVersion: argoproj.io/v1alpha1
 kind: Workflow
 metadata:
   generateName: conditional-
 spec:
-  entrypoint: conditional-example
+  entrypoint: conditionalExample
   arguments:
     parameters:
-    - name: should-print
-      value: "false"
+    - name: shouldPrint
+      value: "true"
 
   templates:
-  - name: conditional-example
+  - name: conditionalExample
     inputs:
       parameters:
-      - name: should-print
+      - name: shouldPrint
```

Comment:
removed the dash from the name since expr expressions which include dash need to be formatted with brackets due to the special character

Reply:
Could you clarify with some examples? Does this mean names with dashes will not work?

Reply:
Thanks for the suggestion. I changed it back to the dashed name to show how you can handle that with expr. 

BTW, [this](https://github.com/argoproj/argo-workflows/blob/master/examples/life-cycle-hooks-tmpl-level.yaml) is the example Bala had, which I imitated.